tests/SourceTest.php

```diff
@@ -2,211 +2,98 @@
-                'city' => "Test City",
-                'postal_code' => "12345",
-                'state' => "Test State",
-                'country' => "Test Country",
-            )
-        );
-        $this->mockRequest(
-            'POST',
-            '/v1/sources/src_foo',
-            array(
-                'owner' => array(
-                    'name' => "Stripey McStripe",
-                    'address' => array(
-                        'line1' => "Test Address",
-                        'city' => "Test City",
-                        'postal_code' => "12345",
-                        'state' => "Test State",
-                        'country' => "Test Country",
-                    ),
-                ),
-            ),
-            $response
-        );
-
-        $source = Source::retrieve('src_foo');
-        $source->owner['name'] = "Stripey McStripe";
-        $source->owner['address'] = array(
-            'line1' => "Test Address",
-            'city' => "Test City",
-            'postal_code' => "12345",
-            'state' => "Test State",
-            'country' => "Test Country",
-        );
-        $source->save();
-        $this->assertSame($source->owner['name'], "Stripey McStripe");
-        $this->assertSame($source->owner['address']['line1'], "Test Address");
-        $this->assertSame($source->owner['address']['city'], "Test City");
-        $this->assertSame($source->owner['address']['postal_code'], "12345");
-        $this->assertSame($source->owner['address']['state'], "Test State");
-        $this->assertSame($source->owner['address']['country'], "Test Country");
+        $resource = Source::update(TEST_RESOURCE_ID, array(
+            "metadata" => array("key" => "value"),
+        ));
+        $this->assertSame("Stripe\\Source", get_class($resource));
     }
 
-    public function testDetachAttached()
+    public function testIsDetachable()
     {
-        $response = array(
-            'id' => 'src_foo',
-            'object' => 'source',
-            'customer' => 'cus_bar',
-        );
-        $source = Source::constructFrom(
-            $response,
-            new Util\RequestOptions()
-        );
-
-        unset($response['customer']);
-        $this->mockRequest(
-            'DELETE',
-            '/v1/customers/cus_bar/sources/src_foo',
-            array(),
-            $response
-        );
-
-        $source->detach();
-        $this->assertFalse(array_key_exists('customer', $source));
+        $resource = Source::retrieve(TEST_RESOURCE_ID);
+        $resource->customer = "cus_123";
+        $this->expectsRequest(
+            'delete',
+            '/v1/customers/cus_123/sources/' . TEST_RESOURCE_ID
+        );
+        $resource->delete();
+        $this->assertSame("Stripe\\Source", get_class($resource));
     }
 
-    /**
-     * @expectedException Stripe\Error\Api
-     */
-    public function testDetachUnattached()
+    /*
+    This does not yet work with our version of PHP Unit?
```

Comment:
@ob-stripe Tried to catch the exception but got errors because `expectException` is not defined. Seems to be part of a different class than the one you inherit from for StripeMockCase. Didn't want to break everything so deferring to you

Reply:
Yes, we're on a stupidly old version of PHPUnit because of PHP 5.3 :yellatcloud:

You can still expect exceptions via a comment with the `@expectedException` tag. Take a look at `WebhookTest.php` for a few examples.

Reply:
Ah cool, I noticed this one but assumed it was unrelated. Fixed in https://github.com/stripe/stripe-php/pull/389/commits/9ff4172531127459320aef1945e69cdc5eec5d7f